Console/Command/ConfigSetCommand.php

```diff
@@ -2,77 +2,111 @@
     public function __construct(
-        \Magento\Framework\App\State $state,
+        State $state,
         Factory $configFactory,
-        \Thai\S3\Helper\S3 $helper
+        S3Helper $helper
     ) {
         $this->state = $state;
         $this->helper = $helper;
         $this->configFactory = $configFactory;
+
         parent::__construct();
     }
 
+    /**
+     * @inheritdoc
+     */
     protected function configure()
     {
         $this->setName('s3:config:set');
         $this->setDescription('Allows you to set your S3 configuration via the CLI.');
         $this->setDefinition($this->getOptionsList());
     }
 
+    /**
+     * {@inheritdoc}
+     *
+     * @throws \Exception
+     */
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        try {
-            $this->state->setAreaCode('adminhtml');
-        } catch (\Magento\Framework\Exception\LocalizedException $e) {
-            // intentionally left empty
-        }
+        $this->state->emulateAreaCode(Area::AREA_ADMINHTML, function () use ($input, $output) {
+            if (!$input->getOption('region') && !$input->getOption('bucket') && !$input->getOption('secret-key') && !$input->getOption('access-key-id')) {
+                $output->writeln($this->getSynopsis());
 
-        if (!$input->getOption('region') && !$input->getOption('bucket') && !$input->getOption('secret-key') && !$input->getOption('access-key-id')) {
-            $output->writeln($this->getSynopsis());
-            return;
-        }
+                return;
+            }
 
-        $errors = $this->validate($input);
-        if ($errors) {
-            $output->writeln('<error>' . implode('</error>' . PHP_EOL .  '<error>', $errors) . '</error>');
-            return;
-        }
+            $errors = $this->validate($input);
+            if ($errors) {
+                $output->writeln('<error>' . implode('</error>' . PHP_EOL . '<error>', $errors) . '</error>');
 
-        $config = $this->configFactory->create();
+                return;
+            }
 
-        foreach ($this->getOptions() as $option => $pathValue) {
-            if ( ! empty($input->getOption($option))) {
-                $config->setDataByPath('thai_s3/general/'.$pathValue, $input->getOption($option));
-                $config->save();
+            $config = $this->configFactory->create();
+
+            foreach ($this->getOptions() as $option => $pathValue) {
+                if (!empty($input->getOption($option))) {
+                    $config->setDataByPath('thai_s3/general/' . $pathValue, $input->getOption($option));
+                    $config->save();
+                }
             }
-        }
 
-        $output->writeln('<info>You have successfully updated your S3 credentials.</info>');
+            $output->writeln('<info>You have successfully updated your S3 credentials.</info>');
+        });
     }
 
+    /**
+     * @return array
+     */
     public function getOptions()
```

Comment:
Should we be type hinting return types as well?

Reply:
It's currently compatible with 5.6, so no type hinting